test/realm_test.dart

```diff
@@ -824,6 +824,71 @@ Future<void> main([List<String>? args]) async {
     expect(stored.location, now.location);
     expect(stored.location.name, 'Europe/Copenhagen');
   });
+
+  test('Realm open with wrong encryption key', () {
```

Comment:
These are all testing that a Realm that is already opened with 1 encryption key cannot be opened. But we're not testing a scenario where we open, then close, then try to open again. You could probably parameterize the tests to avoid having to copy-paste them all.

Reply:
This test covers this scenario "Realm - open local encrypted realm with encryption key". Yes, I could think about parameterizing.

Reply:
The issue is that it doesn't cover the case where the Realm is not open when you provide the invalid encryption key.

Reply:
OK, I see. It is another special case :). I added the test.